components/toolbar/src/api/statements.ts

```diff
@@ -0,0 +1,129 @@
+import { DatabaseAdapter } from 'electric-sql/electric'
+import { DbTableInfo, TableColumn } from './interface'
+
+export type SqlDialect = 'sqlite' | 'postgres'
+
+const SQLITE_GET_TABLES = `
+SELECT name, sql FROM sqlite_master
+WHERE type='table'
+AND name NOT LIKE 'sqlite_%'
+`
+
+const SQLITE_GET_DB_TABLES = `
+${SQLITE_GET_TABLES}
+AND name NOT LIKE '_electric_%'
+`
+
+const SQLITE_GET_ELECTRIC_TABLES = `
+${SQLITE_GET_TABLES}
+AND name LIKE '_electric_%'
+`
+
+const PG_GET_TABLES = `
+SELECT table_name AS name FROM information_schema.tables
+WHERE table_schema='public'
+AND table_name NOT LIKE 'pg_%'
+`
+const PG_GET_DB_TABLES = `
+${PG_GET_TABLES}
+AND table_name NOT LIKE '_electric_%'
+`
+
+const PG_GET_ELECTRIC_TABLES = `
+${PG_GET_TABLES}
+AND table_name LIKE '_electric_%'
+`
+
+export const getSqlDialect = async (
+  adapter: DatabaseAdapter,
+): Promise<SqlDialect> => {
+  try {
+    await adapter.query({
+      sql: `SELECT 1 FROM sqlite_master;`,
+    })
+    return 'sqlite'
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (err: any) {
+    if (
+      err &&
+      !err.message.includes('relation "sqlite_master" does not exist')
+    ) {
+      throw err
+    }
+  }
+  try {
+    await adapter.query({
+      sql: `SELECT 1 FROM information_schema.tables;`,
```

Comment:
`information_schema` is in the SQL standard, not a PG specific thing, and so SQLite could adopt it in future, would be better to use `pg_catalog.pg_class`.

```suggestion
      sql: `SELECT 1 FROM pg_catalog.pg_class;`,
```